src/cpp/ext/gcp_observability/gcp_observability.cc

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <grpc/support/port_platform.h>
+
+#include "src/cpp/ext/gcp_observability/gcp_observability.h"
+
+#include <stdint.h>
+
+#include <utility>
+
+#include "opencensus/exporters/stats/stackdriver/stackdriver_exporter.h"
+#include "opencensus/exporters/trace/stackdriver/stackdriver_exporter.h"
+#include "opencensus/trace/sampler.h"
+#include "opencensus/trace/trace_config.h"
+
+#include <grpcpp/opencensus.h>
+
+namespace grpc {
+namespace experimental {
+
+namespace {
+// TODO(yashykt): These constants are currently derived from the example at
+// https://cloud.google.com/traffic-director/docs/observability-proxyless#c++.
+// We might want these to be configurable.
```

Comment:
@DNVindhya we don't configure these in Java right? Why are there max values for attrs, annotations, events etc? Do we set them to the same values i.e. 128 for all 3?

Reply:
Yes, in Java we only configure the trace sampler based on Observability Config and for rest we use [default](https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/trace/config/TraceParams.java#L50) values.

Reply:
So only events value is 128 and the other 2 are 32 each. Ideally we should have the same default values for all languages but right now it's not a blocker for this PR